### PR TITLE
Power Up Creation

### LINKED ISSUE #67: Power-ups
The project brief for the game requires the chef to be able to obtain power ups during the course of their gameplay. The method for obtaining these will be purchasing them in the same shop where a player can buy new chefs or stations. We are required to implement 5 total power ups, which are detailed as follows:

1. Chef Speed Up (increases the speed of the chefs)
2. Preparation Speed Up (reduces the time taken to prepare an ingredient, such as at the grill, oven or chopping board.)
3. No Failing Preparation (prevents preparation steps from being failed whilst the power up is active, e.g. a patty will not burn whilst the power up is active.)
4. No Reputation Loss (prevents the player losing reputation whilst the power up is active.)
5. More Money (increases the amount of money a player will get from completing an order. This will require the money functionality to be implemented, as described in #66.)

### PR DESCRIPTION
**Pull request by roh518**
Closes #67

### CODE REVIEW

File: core/src/cs/eng1/piazzapanic/PlayerState.java [hunk 1]
```diff
@@ -1,32 +1,59 @@
...
+
+    private Timer[] powerUpTimers = {
+        new Timer(10000, true, false), // doubleChefSpeed
+        new Timer(60000, false, false), // doublePrepSpeed
+        new Timer(60000, false, false), // noFailPrep
+        new Timer(60000, false, false), // noRepLoss
+        new Timer(60000, false, false), // moreMoney
+    };
+
+    public static PlayerState getInstance() {
```
Comment (dislocated-su) [inline core/src/cs/eng1/piazzapanic/PlayerState.java:21]: Would the Singleton pattern here mean that PlayerState can carry over multiple games? Like if I exit scenario mode and then enter endless?

File: core/src/cs/eng1/piazzapanic/screens/GameScreen.java [hunk 2]
```diff
@@ -41,6 +41,7 @@ public class GameScreen implements Screen {
     private final UIOverlay uiOverlay;
     private final FoodTextureManager foodTextureManager;
     private final CustomerManager customerManager;
+    // private final PowerUps powerUps;
```
Comment (dislocated-su) [inline core/src/cs/eng1/piazzapanic/screens/GameScreen.java:44]: Probably want to remove that

File: core/src/cs/eng1/piazzapanic/screens/GameScreen.java [hunk 3]
```diff
@@ -73,14 +74,17 @@ public GameScreen(final PiazzaPanicGame game, int totalCustomers) {
 
         foodTextureManager = new FoodTextureManager();
 
+        PlayerState.reset();
+
+        // powerUps = new PowerUps();
```
Comment (dislocated-su) [inline core/src/cs/eng1/piazzapanic/screens/GameScreen.java:79]:
```suggestion
```

**Review by dislocated-su — CHANGES_REQUESTED** [review]
Haven't tested yet but had some questions while reading the code

File: core/src/cs/eng1/piazzapanic/PlayerState.java (hunk 1 above)
Comment (dislocated-su) [reply core/src/cs/eng1/piazzapanic/PlayerState.java:21]: I can that the PlayerState is reset.

**Review by dislocated-su — APPROVED** [review]

**Review by LouisWarren — APPROVED** [review]
lgtm

**Pull request merged**